Lesson14

AndreyPushai opened the pull request.

AndreyPushai committed:
Upd package

Added typescript and jest to dependencies

AndreyPushai committed:
Upd package.json

- Upd scripts
- Upd dependencies

AndreyPushai committed:
Added jest.config.ts

AndreyPushai committed:
Added home work for lesson14

apolonskiy commented on lesson14/api.test.ts:
```diff
@@ -0,0 +1,149 @@
+
+interface PayloadObject {
+    name: string;
+    data: object | string;
```
дату теж далі можна описувати вкладеним обєктом

AndreyPushai replied:
В дату передаються обʼєкти з довільними ключами

apolonskiy replied:
Ага, зрозумів

apolonskiy commented on lesson14/api.test.ts:
```diff
@@ -0,0 +1,149 @@
+    data: object | string;
+};
+
+const axiosInstance: AxiosInstance = axios.create({
```
я б очікував що ви клас напишете, але якщо так то немаю питань)

AndreyPushai replied:
Done.

AndreyPushai committed:
Upd eslint.config.js

- Added configs for jest.
- Added dist dir to ignorelist.

AndreyPushai committed:
Upd jest.config.ts

Removed inappropriate configs

AndreyPushai committed:
Upd lesson14 homework

- Replaced axiosInstance and async API functions with AxiosHandler class
wrapper
- Added axiosHandler instances to every test suit

AndreyPushai committed:
Upd lesson 14 homework

Added negative tests
